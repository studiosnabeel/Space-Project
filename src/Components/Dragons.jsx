import React from 'react'

export const Dragons = () => {
  return (
    <div>Dragons</div>
  )
}
