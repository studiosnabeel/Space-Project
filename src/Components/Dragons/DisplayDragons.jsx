import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Dragons from './Dragons';
import { selectDragons, fetchDragonApi } from '../../Redux/dragonSlice';

const DisplayDragons = () => {
  const dragons = useSelector(selectDragons);
  const dispatch = useDispatch();
  useEffect(() => {
    if (dragons.length === 0) {
      dispatch(fetchDragonApi());
    }
  }, [dispatch, dragons.length]);

  return (
    <div>
      <div>
        {dragons.map((dragon) => (
          // eslint-disable-next-line react/jsx-props-no-spreading
          <Dragons key={dragon.id} {...dragon} />
        ))}
      </div>
    </div>
  );
};

export default DisplayDragons;
