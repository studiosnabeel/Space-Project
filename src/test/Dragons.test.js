import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Dragons from '../Components/Dragons/Dragons';

jest.mock('react-redux');

describe('Rockets test', () => {
  test('should test rocket component', () => {
    useSelector.mockReturnValue([
      {
        id: 'dragon1',
        flicker_images: 'https://i.imgur.com/9fWdwNv.jpg',
        rocket_name: 'Dragon 1',
        description: 'Test',
        reserved: !true,
      },
    ]);
    const tree = render(<Dragons />);
    expect(tree).toMatchSnapshot();
  });
});
