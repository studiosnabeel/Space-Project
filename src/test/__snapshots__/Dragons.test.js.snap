// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets test should test rocket component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-2 mb-4 flex mx-12 items-start"
      >
        <div
          class=" flex items-center w-[200px] h-[170px] lg:w-[400px] lg:h-[300px] flex-none"
        >
          <img
            alt="picture1"
            class="self-center object-contain w-[400px] h-[300px]"
          />
        </div>
        <div
          class=" mx-2"
        >
          <h3
            class="font-bold text-lg"
          />
          <p />
          <button
            class="bg-[#007bff] text-white px-3 py-1.5 rounded mt-2 "
            type="button"
          >
            Reserve rocket
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-2 mb-4 flex mx-12 items-start"
    >
      <div
        class=" flex items-center w-[200px] h-[170px] lg:w-[400px] lg:h-[300px] flex-none"
      >
        <img
          alt="picture1"
          class="self-center object-contain w-[400px] h-[300px]"
        />
      </div>
      <div
        class=" mx-2"
      >
        <h3
          class="font-bold text-lg"
        />
        <p />
        <button
          class="bg-[#007bff] text-white px-3 py-1.5 rounded mt-2 "
          type="button"
        >
          Reserve rocket
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
